tests/mod_learningmap_mapworker_test.php

```diff
@@ -0,0 +1,98 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace mod_learningmap;
+
+/**
+ * Unit test for mod_learningmap
+ *
+ * @package     mod_learningmap
+ * @copyright   2021-2022, ISB Bayern
+ * @author      Stefan Hanauska <stefan.hanauska@csg-in.de>
+ * @license     http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ * @group      mod_learningmap
+ * @group      mebis
+ * @covers     \mod_learningmap\mapworker
+ */
+class mod_learningmap_mapworker_test extends \advanced_testcase {
+
+    /**
+     * Prepare testing environment
+     */
+    public function setUp(): void {
+        global $DB;
+        $this->course = $this->getDataGenerator()->create_course(['enablecompletion' => 1]);
+        $this->learningmap = $this->getDataGenerator()->create_module('learningmap',
+            ['course' => $this->course, 'completion' => 2, 'completiontype' => 0]);
+
+        $this->activities = [];
+        for ($i = 0; $i < 9; $i++) {
+            $this->activities[] = $this->getDataGenerator()->create_module(
+                'page',
+                ['name' => 'A', 'content' => 'B', 'course' => $this->course, 'completion' => 2, 'completionview' => 1]
```

Comment:
same here. If not, maybe a comment, what kind of completion setting is being applied here, would be nice